internal/pithosctl/pkg/pithos/healthz.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright (C) 2020 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pithos
+
+import (
+	"bytes"
+
+	"github.com/gravitational/pithos-app/internal/pithosctl/pkg/defaults"
+
+	"github.com/gravitational/trace"
+	minio "github.com/minio/minio-go"
+)
+
+const (
+	defaultBucketLocation = "G1"
+	objectName            = "liveness-probe.txt"
+)
+
+// HealthzConfig defines access configuration to pithos server for health checks
+type HealthzConfig struct {
+	// AccessKey defines pithos S3 access key
+	AccessKey string
+	// SecretAccessKey defines pithos S3 secret key
+	SecretAccessKey string
+	// Endpoint defines pithos endpoint address
+	Endpoint string
+}
+
+// Check checks configuration parameters
+func (h *HealthzConfig) CheckAndSetDefaults() error {
+	if h == nil {
+		return nil
+	}
+
+	var errors []error
+	if h.AccessKey == "" {
+		errors = append(errors, trace.BadParameter("access-key is required"))
+	}
+	if h.SecretAccessKey == "" {
+		errors = append(errors, trace.BadParameter("secret-access-key is required"))
+	}
+	if h.Endpoint == "" {
+		h.Endpoint = defaults.HealthzEndpoint
+	}
+	return trace.NewAggregate(errors...)
+}
+
+// Healthz defines configuration for pithos server health checks
+type Healthz struct {
+	// Bucket defines bucket used for health checks
+	Bucket string
+	// Client represent minio.io S3 client
+	Client *minio.Client
+}
+
+func NewHealthz(config HealthzConfig, bucket string) (*Healthz, error) {
+	if bucket == "" {
+		bucket = defaults.HealthzBucket
+	}
+
+	client, err := initClient(config)
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to create s3 client")
+	}
+
+	return &Healthz{
+		Bucket: bucket,
+		Client: client,
+	}, nil
+}
+
+func initClient(config HealthzConfig) (*minio.Client, error) {
+	insecure := false
+	client, err := minio.NewV2(config.Endpoint, config.AccessKey, config.SecretAccessKey, insecure)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return client, nil
+}
+
+// Prepare creates bucket and object if they are not exist
```

Comment:
`if they do not exist`
Will it fail if the bucket/object exist?

Reply:
Did not check it in my testing. Will do it today.

Reply:
In that case, PUT is used instead of POST, which makes Prepare funcion save to use on multiple pods.